### ISSUE TITLE
Build baryon challenge infrastructure

### ISSUE THREAD

**Issue by tilmantroester** [LSSTDESC/SprintWeek2023#14]
# Baryon challenge

We need to build the infrastructure to create and analyse mock data vectors for the baryon challenge. 

Contacts: @tilmantroester, @roohi-dalal, @aamon, @elisachisari 
Day/Time: ~all week, remote
Main communication channel: #desc-baryon-challenge
GitHub repo:
Zoom room: https://stanford.zoom.us/j/99864191818?pwd=eVJiZ0c3ODBWK1ljcVBjcFJGdTFKUT09

## Goals and deliverable

1. Implement a P(k, z) modifier systematic in firecrown
2. Build an API that takes in a power spectrum model and scale cuts and returns a firecrown likelihood specification
2.1 Stretch goal: use API to create mock data vector with specified baryon model
3. Decide on which mocks to use and if/how to blind
4. Finalise the evaluation requirements

## Resources and skills needed

firecrown, CCL, sacc

## Detailed description

For the baryon challenge we need an API that allows entrants to easily plug their model for the matter power spectrum into a firecrown likelihood. The goal of this sprint is to define this API and ideally implement it. This might require changes to the guts of firecrown. Once the API is in place, it can also be used to create the mock data vectors for the challenge, which would be the stretch goal for the week.

**Comment by tilmantroester** [comment]
A draft implementation of a P(k,z) modifier systematic in firecrown: https://github.com/LSSTDESC/firecrown/pull/329
